Foundation/src/DateTimeParser.cpp

```diff
@@ -42,10 +43,19 @@ namespace Poco {
 	{ int i = 0; while (i < n && it != end && Ascii::isDigit(*it)) { var = var*10 + ((*it++) - '0'); i++; } while (i++ < n) var *= 10; }
 
 
-void DateTimeParser::parse(const std::string& fmt, const std::string& str, DateTime& dateTime, int& timeZoneDifferential)
+inline void cleanupInputString(std::string& str)
+{
+	Poco::trimInPlace(str);
+}
+
+void DateTimeParser::parse(const std::string& fmt, const std::string& dtStr, DateTime& dateTime, int& timeZoneDifferential, bool cleanup)
 {
+	std::string str(dtStr);
```

Comment:
to avoid string copy here, this may be implemented using `string_view` into `find_first_not_of` and `find_last_not_of` of the original string

Reply:
It is not easy to avoid copying without major changes in the parser to use string_view instead (or in addition to string).